### DISCUSSION TITLE
Watlow F4

### DISCUSSION THREAD

**Discussion by Dgocof** [artisan-roaster-scope/artisan#1000] [General]
I was hoping to get some help setting up a Watlow F4 to work with Artisan. I have run wires from connections 12,13,com. to an RS485 converter usb but cannot navigate the software setup and need help.

**Comment by poundy** [comment]
you mean this one? https://www.watlow.com/Products/Controllers/Temperature-and-Process-Controllers/SERIES-F4-Process-Controller

Things you will need to look at are to look at the device serial port settings and configure your RS485 connection to that, and then look at the register map and find what settings you need in the modbus page.

**Reply by Dgocof** [reply]
Yes that is the controller. I'm wondering if the RS485 needs device software because when I plug it into the computer it doesn't show up in device manager in windows when I plug it in. Usually windows recognizes a new device and installs software for it. The other settings you mentioned are greek to me. I was hoping to find someone to walk me through some of this stuff. Thanks

**Reply by Dgocof** [reply]
What is the register map and where is that located? It appears the Rs485 is installed and on com 4. 
Should I choose a machine? Even though they don't have mine.
Under Device I chose device and set it to meter with baud rate of 9600.
Can I use PID? I don't have a fuji device. Eventually I would like this to control my Watlow with a profile.
Do I configure a port? I directed it to Com 4. and Modbus to com 4

**Reply by poundy** [reply]
ignoring this thread

**Comment by bigl00z3** [comment]
@Dgocof,
In between your Watlow F4 and your computer, you have the serial to usb converter. This is the device that may need a driver. This highly depends on the model. Which one are you using?

Just as examples, I have used 3 models (they are many) that didn't require any additional drivers (thanks to the widely supported FTDI chip). I have used them without any issues on Mac, Windows and Raspberry Pi:

- [The US Converter USB to RS485 / RS422 Converter, Mini Premium (XS885)](https://www.usconverters.com/usb-rs485-converter-xs885)
- [The Joy IT SBC-TTL-RS485](https://joy-it.net/en/products/SBC-TTL-RS485)  (<-This one is cheaper but it's a lighter build)
- [EX-1309-T](https://www.distrelec.ch/en/usb-serial-converter-rs232-rs422-rs485-terminal-block-exsys-ex-1309/p/30082283?queryFromSuggest=true)

Also:

- 1st check in your computer device manager that it is visible. Artisan won't communicate if in the first place the computer can't see it.
- Once the computer sees it, make sure that the settings (Baud rat, Byte Size, Parity...) are exactly the same on both the hardware (Watlow) and software (Artisan).

I hope this makes sense.

Nico

**Reply by Dgocof** [reply]
[DFROBOT](https://www.newark.com/b/dfrobot)
Manufacturer Part No:	FIT0737
Newark Part No.:	52AJ5098
This is the usb part. Usually windows recognizes a new device and finds a driver but not this time.

**Reply by Dgocof** [reply]
Thanks for the links. I'll look at another usb maybe.

**Reply by bigl00z3** [reply]
Looking at the [DFRobot website](https://www.dfrobot.com/product-2189.html) for the device, there is an installer you apparently can download under Documets [Drive](http://data.ic5188.com/PDF/SC-US_PL2303.rar)

After installing the driver, do you see the device in your computer Device Manager?

Nico

**Reply by Dgocof** [reply]
I'll give this a try when we finish roasting today.

**Reply by Dgocof** [reply]
Drive does nothing when I tried it. I can see the device in the device manager on com 4 with a driver installed. It appears the driver is installed on com 4 but still no temp showing. 
Should I choose a machine? Even though they don't have mine. 
Under Device I chose device and set it to meter with baud rate of 9600. 
Can I use PID? I don't have a fuji device. Eventually I would like this to control my Watlow with a profile. 
Do I configure a port? I directed it to Com 4. and Modbus to com 4

**Reply by poundy** [reply]
you're replying in multiple places, and just cluttering the conversation. Lets keep working here.

If you choose a machine, what are you going to choose? None of them will match what hardware you have. As it's your device, you are possibly the only person who has it, nobody else can "debug" this as easily as you. So no, don't set a device, except possibly to look at the settings that a particular machine uses - there's a very high chance that none of the presets will work with your equipment, but looking at the dialog boxes to see what kind of detail you need to complete is OK (just remember to reset the configuration before setting it up for your hardware).

What you need to do now is set up two things in Artisan... First the serial port must be configured how your USB adapter and PID expect. Read the datasheet to find that. Second, Artisan needs to know what registers correspond to it's data fields. Again, that's the register map, as documented in the Watlow datasheet. Those datasheets are your homework...

**Reply by Dgocof** [reply]
I believe the F4 is communicating on 19200 baud because when I scan under device, modbus it returns parameters in the f4. When I go back to BT/ET I set it to comm 4 ,19200 but still don't see temps, same as modbus? What else should I look at? Thanks

**Reply by bigl00z3** [reply]
On the [web site](https://www.watlow.com/Products/Controllers/Temperature-and-Process-Controllers/SERIES-F4-Process-Controller), if you look at the [user manual ](https://www.watlow.com/-/media/documents/user-manuals/f4pe-e.ashx)for your device:

Page 46:
![image](https://user-images.githubusercontent.com/8886922/199827939-b8f0e6c3-f1f4-488a-9c2d-010e7fec583e.png)

While the default is mentionned as 19200, you should always check on your device what is set and then replicate in Artisan. 
In an example for my Fuji PXF.
![image](https://user-images.githubusercontent.com/8886922/199829037-eace38a6-39b1-4bce-b056-fc8f427def48.png)

Each line (not only the baud rate) of the config is important. You need the exact combination, otherwise Artisan will not be able to read it.

Then you also need to access the right register and settings for the parameter you are looking for. 
As an example on the Fuji PXF:
![image](https://user-images.githubusercontent.com/8886922/199830355-67043f84-f7b8-40e7-b761-1a1f3df8a11c.png)
![image](https://user-images.githubusercontent.com/8886922/199830395-ebe8c0de-c5da-4daa-bda6-08e4d228da62.png)


In your device's user manual, check Chapter 9, page 95 and further:
![image](https://user-images.githubusercontent.com/8886922/199830044-be461199-2b96-4c3f-99be-c344db259bfd.png)

If you want examples of PID settings 8on other MODBUS devices,  you can check [here](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-fuji-pxf-and-test-of-maxwell-mta) and [here](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-using-bt-as-input-fuji-pxf) for my setup using a Fuji PXF or also on [Frans's site](https://kostverlorenvaart.blogspot.com/search?q=PID). 

Nico

**Reply by Dgocof** [reply]
Does the modbus need to be setup before I will see a temperature in the software or just the bt/et ?

**Reply by poundy** [reply]
If you can't talk to the Watlow, then Artisan will not have any data to display. Zero. If you don't get the serial port set up correct, you can't talk to the Watlow. If you can't get the correct registers set up in the modbus dialog, you get no data.

**Comment by Dgocof** [comment]
Hi Nico, So I found out that the Watlow f4 software and Artisan were conflicting. The equipment was hooked up correctly but the temperature reading would swap back and forth between the Watlow controlling software and Artisan.  Now we are using Phidgets to record our existing profiles. My question is I'm thinking of using the Fiji controller. It appears it is set up for an air roaster as it monitors and controls ET. Do you get around that with a drum roaster by swapping the ET BT? Thanks
